pkg/domain/infra/abi/containers.go

```diff
@@ -77,10 +78,31 @@ func getContainersAndInputByContext(all, latest bool, names []string, filters ma
 			}
 		}
 	case latest:
-		ctr, err = runtime.GetLatestContainer()
-		if err == nil {
-			rawInput = append(rawInput, ctr.ID())
-			ctrs = append(ctrs, ctr)
+		isPod := false
+		for _, name := range names {
+			obj := regexp.MustCompile("^[a-zA-Z0-9_.-]{12}-infra")
```

Comment:
This assumption is not true at all, you can change the infra name to anything. But more importantly if --latest is set you are not allowed to specify names so the loop is never used. I don't understand how this can work at all right now?